kb/migrating-visibility-data-from-es6.md

```diff
@@ -0,0 +1,82 @@
+---
+slug: migrate-visibility-data-from-es6
+title: Migrate visibility data from ES6
+tags:
+  - kb-article
+date: 2022-29-08T00:00:00Z
+---
+
+We added support for [Elasticsearch v7+ (ES7)](https://www.elastic.co/downloads/past-releases/#elasticsearch) in the v1.7.0 update to the [Temporal Server](/clusters#temporal-server).
+ES7 introduces several breaking changes, including the removal of mapping types.
+These changes make ES6 incompatible with ES7.
+
+Upgrade ES6 to ES7, and then use one of the following methods to update your Temporal Server.
+
+:::note
+
+If you are using AWS Elasticsearch, only upgrade to version 7.7.
+
+:::
+
+## Fresh installation
+
+1. Install the latest version of Temporal Server.
+2. Install ES7 as shown in the [Elasticsearch documentation](https://www.elastic.co/guide/en/elasticsearch/reference/current/setup-upgrade.html).
+3. To set the Elasticsearch version in Temporal Server:
+
+- Set the version in the server's static config file:
+
+  ```
+  persistence:
+    datastores:
+      es-visibility:
+        elasticsearch:
+          version: "v7"
+  ```
+
+- Set `ES_VERSION` to `v7` in the pre-built Docker image.
+
+## Upgrade with temporary shutdown
+
+1. Shut down the Temporal Cluster.
+2. Upgrade ES6 to ES7 according to [Elasticsearch documentation](https://www.elastic.co/guide/en/elasticsearch/reference/current/setup-upgrade.html).
+
+## Rolling upgrade
+
```

Comment:
Here what it is.

1. If you are still using Elasticsearch v6 make sure you have 
```
 persistence:
   datastores:
     es-visibility:
       elasticsearch:
         version: "v6"
```
in your config file or set `ES_VERSION` env to `v6` if you are using pre-build docker image.
2. Update Temporal to latest version but bellow 1.18 (i.e. 1.17.5)
3. Add the following to the dynamic config:
 ```
 history.visibilityTaskWorkerCount:
   - value: 0
 ```
This will disable internal visibility queue processor.
4. your (3)
5. your (4)
6. your (5)
7. Remove 
```
 persistence:
   datastores:
     es-visibility:
       elasticsearch:
         version: "v6"
```
from you config file or `ES_VERSION` env if you are using pre-build docker image.
8. Remove this from the dynamic config:
```
 history.visibilityTaskWorkerCount:
   - value: 0
```
9. your (9)